python/ray/serve/scripts.py

```diff
@@ -297,71 +215,77 @@ def deploy(config_file_name: str, address: str):
-    args, kwargs = _process_args_and_kwargs(args_and_kwargs)
-
-    # Calculate deployments' runtime env updates requested via args
-    runtime_env_updates = parse_runtime_env_args(
+    final_runtime_env = parse_runtime_env_args(
         runtime_env=runtime_env,
         runtime_env_json=runtime_env_json,
         working_dir=working_dir,
     )
 
-    # Create ray.init()'s runtime_env
-    if "working_dir" in runtime_env_updates:
-        ray_runtime_env = {"working_dir": runtime_env_updates.pop("working_dir")}
-    else:
-        ray_runtime_env = {}
-
-    if is_config:
+    app_or_node = None
+    if pathlib.Path(config_or_import_path).is_file():
         config_path = config_or_import_path
-        # Delay serve.start() to catch invalid inputs without waiting
-        if len(args) + len(kwargs) > 0:
-            raise ValueError(
-                "ARGS_AND_KWARGS cannot be defined for a "
-                "config file deployment. Please specify the "
-                "init_args and init_kwargs inside the config file."
-            )
-
-        cli_logger.print("Deploying application in config file at " f"{config_path}.")
+        cli_logger.print(f"Loading app from config file: '{config_path}'.")
         with open(config_path, "r") as config_file:
-            app = Application.from_yaml(config_file)
-
+            app_or_node = Application.from_yaml(config_file)
     else:
         import_path = config_or_import_path
         if "." not in import_path:
             raise ValueError(
                 "Import paths must be of the form "
-                '"module.submodule_1...submodule_n.MyClassOrFunction".'
+                "'module.submodule.app_or_bound_deployment'."
             )
 
-        cli_logger.print(f'Deploying function or class imported from "{import_path}".')
-
-        deployment_name = import_path[import_path.rfind(".") + 1 :]
-        deployment = serve.deployment(name=deployment_name)(import_path)
+        cli_logger.print(f"Loading app from import path: '{import_path}'.")
+        app_or_node = import_attr(import_path)
 
-        app = Application([deployment.options(init_args=args, init_kwargs=kwargs)])
+    # Setting the runtime_env here will set defaults for the deployments.
+    ray.init(address=address, namespace="serve", runtime_env=final_runtime_env)
 
-    ray.init(address=address, namespace="serve", runtime_env=ray_runtime_env)
-
-    for deployment in app:
-        _configure_runtime_env(deployment, runtime_env_updates)
-
-    app.run(logger=cli_logger)
+    try:
+        serve.run(app_or_node, host=host, port=port)
+        cli_logger.success("Application deployed successfully.")
+        if blocking:
+            cli_logger.print("Watching deployment statuses:\n")
+            while True:
+                print(get_deployment_statuses())
+                time.sleep(10)
+    except KeyboardInterrupt:
+        cli_logger.info("Got KeyboardInterrupt), shutting down...")
+        serve.shutdown()
```

Comment:
Removing the deployment deletion logic will cause `test_run_simultaneous` in `test_cli.py` to fail.

The test simulates a user opening two `serve run` sessions at once. When they close one session, they should still have access to deployments from the other one, but calling `serve.shutdown()` here will remove all the deployments.